### PR TITLE
Use prettyprinter-1.7.0

### PR DESCRIPTION
**Pull request by sjakobi**
Changelog:
http://hackage.haskell.org/package/prettyprinter-1.7.0/changelog

hnix is likely to be affected by this change:

> Use floor instead of round to compute ribbon width.

…due to its use of custom `LayoutOptions`.

### CODE REVIEW

**Comment by sjakobi** [comment]
I'm not super happy about adding another tight bound on a dependency, but I think it's somewhat justified here:

1. 

> hnix is likely to be affected by this change:
> 
> > Use floor instead of round to compute ribbon width.
> 
> …due to its use of custom `LayoutOptions`.

The use of `LayoutOptions` with a fractional ribbon width is actually pretty limited:

https://github.com/haskell-nix/hnix/blob/778edf2c38b92f638b793a3a66a7735e46747b0f/main/Main.hs#L135-L141

https://github.com/haskell-nix/hnix/blob/778edf2c38b92f638b793a3a66a7735e46747b0f/tests/ParserTests.hs#L394-L402

In `main` this is the central pretty-printing functionality IIUC. We should probably guarantee a consistent formatting there.

2. We benefit from the new shorter module names.

---

@sorki, @Anton-Latukha, could you review?

**Comment by Anton-Latukha** [comment]
Nice, good fix & update.

I see, you yourself made quite a round: https://github.com/quchen/prettyprinter/pull/160, impressive.

We already addressed that let's make the ribbon width natural for now: https://github.com/haskell-nix/hnix/issues/688

From the code `floor` change is obviously a cosmetic for the project.

---

HNix output, as of any good utility - gives information and keeps consistency by the syntax tree of output, not by the number of spaces put.

Thank you.

**Comment by Anton-Latukha** [comment]
Can you comment on the: https://github.com/quchen/prettyprinter/pull/174

I thought there was a `hierarchical naming scheme` for everyone for modules, but I see it is old news, (I've googled around), what is a modern process/stance on it in the Haskell ecosystem?

**Comment by sorki** [comment]
I like the short module names, had the same issue with remembering the long ones :+1:

**Comment by sjakobi** [comment]
> Can you comment on the: [quchen/prettyprinter#174](https://github.com/quchen/prettyprinter/pull/174)
> 
> I thought there was a `hierarchical naming scheme` for everyone for modules, but I see it is old news, (I've googled around), what is a modern process/stance on it in the Haskell ecosystem?

My impression is that many packages currently tend to use short module names that correspond to the package names. E.g. `StmContainers.*` for the [`stm-containers` package](http://hackage.haskell.org/package/stm-containers). Top-level modules like `Data` or `Control` don't have much informational value IMHO.

**Comment by Anton-Latukha** [comment]
TL;DR: The cataloging system is forced and directly that is what makes it too rigid not allows to evolve and move stuff around (and forcing the structure destroys the ways of alternative approaches and solutions), `hierarchical naming scheme` should be optionally used Haskell module structure.

---

I believe the hierarchical naming scheme, despite it simple solution and current realization problems - was/is very beneficial for the ecosystem, especially before the advent of Google. And for a organization approach, discovering modules, remembering module names, and for offline work (download structure subbranches and their docs to compare and use chosen ones).

The person that searches particular functionality - does not know how the project is named, like what `vinyl` project is about, audio processing? People can go into module hierarchy sections to see what is available for the theme, and what are the default practices by default names.

A combination of graph-like cross-linkage (code (deps) and reverse-deps), tagging, and catalog structure - is a very efficient versatile representation of complex modular structures. Graph-like cross-linkage and tagging is sometimes not enough - to organize knowledge, catalog and recommend the best tools.

History goes in spirals, structure centralization problems replaced by ad-hock, decentralization, federalization and back. Haskell structuring is not ideal, we definitely need to free from the forced structure, and destruction of the structure is the antithesis and what would follow is an inevitable process for regrouping.

I believe the catalog structure is not fit to be forced on all modules, it should be optional infrastructure catalog managed by a Haskell group that by the structure provides and directs people towards recommended best modules, default tools to use. Migrating the projects behind default names, when one module technologically superseded another.

For example, so we can decide to use `hierarchical naming scheme` for the core project, or its most computationally demanding parts, so we do not need to track news of Haskell ecosystem and get notified by the structure when we better migrate project to the better modules for the job that become stable and powerful enough that they superceeded the former branch of the catalog.
